lib/src/main/kotlin/net/dean/jraw/references/PublicContributionReference.kt

```diff
@@ -202,4 +203,62 @@ abstract class PublicContributionReference internal constructor(reddit: RedditCl
                 ))
         }
     }
+
+    /**
+     * Sets the [spoilerStatus] of a post using the official Reddit spoiler flair.
+     *
+     * TODO - Requires mod privileges for the subreddit that contains the post.
+     */
+    @EndpointImplementation(Endpoint.POST_SPOILER, Endpoint.POST_UNSPOILER)
+    fun setPostSpoiler(spoilerStatus: Boolean) {
+        val endpoint = if (spoilerStatus) Endpoint.POST_SPOILER else Endpoint.POST_UNSPOILER
+        reddit.request {
+            it.endpoint(endpoint)
+                .post(mapOf(
+                    "id" to fullName
+                ))
+        }
+    }
+
+    /**
+     * Set or unset the announcement [state] of a post in its subreddit.
+     * Reddit provides 2 allowable announcement "slots" for a subreddit, often referred to as sticky posts.
+     *
+     * The [num] argument is optional, an integer between 1 and 3, and only used when stickying a post.
+     * It allows specifying a particular "slot" to sticky the post into,
+     * and if there is already a post stickied in that slot it will be replaced.
+     * If there is no post in the specified slot to replace, or num is None, the bottom-most slot will be used.
+     *
+     * Setting [toProfile] will sticky the post to an authenticated user's profile page.
+     * This is only used with new profile pages, not supported for legacy profiles.
+     *
+     * TODO - Requires mod privileges for the subreddit that contains the post.
+     */
+    @EndpointImplementation(Endpoint.POST_SET_SUBREDDIT_STICKY)
+    fun stickyPost(state: Boolean, num : Int = -1, toProfile: Boolean = false) {
+        if (state && num > 3)
+            throw IllegalArgumentException("$num is more than the allowable announcement slots")
+
+        if (num != -1) {
+            reddit.request {
+                it.endpoint(Endpoint.POST_SET_SUBREDDIT_STICKY)
+                    .post(mapOf(
+                        "id" to fullName,
+                        "num" to num.toString(),
+                        "state" to state.toString(),
+                        "to_profile" to toProfile.toString()
+                    ))
+            }
+        } else {
+            // Leave the optional num value out
+            reddit.request {
+                it.endpoint(Endpoint.POST_SET_SUBREDDIT_STICKY)
+                    .post(mapOf(
+                        "id" to fullName,
+                        "state" to state.toString(),
+                        "to_profile" to toProfile.toString()
+                    ))
+            }
+        }
```

Comment:
This could be a little DRYer:

```kt
val payload = mutableMapOf(
    "id" to fullName,
    "state" to state.toString(),
    "to_profile" to toProfile.toString()
)

if (num < 0) {
    payload.put("num", num.toString())
}

reddit.request {
    it.endpoint(...).post(payload)
}
```